kernel/sched.c

```diff
@@ -36,7 +36,7 @@ struct k_spinlock _sched_spinlock;
 __incoherent struct k_thread _thread_dummy;
 
 static ALWAYS_INLINE void update_cache(int preempt_ok);
-static void halt_thread(struct k_thread *thread, uint8_t new_state);
+static ALWAYS_INLINE void halt_thread(struct k_thread *thread, uint8_t new_state);
```

Comment:
If it's always going to be inlined, it shouldn't need a prototype (which, obviously, can't be inlined per the language spec, though in the post-LTO world many compilers are able to do so).  Take this out, and if it doesn't build let's fix the declaration order.

Reply:
In this case, both they are serving as forward declarations to work around the fact that each calls the other.